src/server/services/analyticsLogger.ts

```diff
@@ -13,25 +13,30 @@ export interface AnalyticsLogger {
    */
   logRedirectAnalytics: (
     req: Express.Request,
-    res: Express.Response,
     shortUrl: string,
     longUrl: string,
   ) => void
+
+  generateCookie: (
+    cookie?: string,
+  ) => [string, string, { maxAge: number }] | null
 }
 
 @injectable()
 export class GaLogger implements AnalyticsLogger {
   logRedirectAnalytics: (
     req: Express.Request,
```

Comment:
can we refactor out the code smell with the need for `req` here?

Reply:
Given the turnaround time, let's hold off on that. I can refactor this after the release